modules/gapi/test/infer/gapi_infer_onnx_test.cpp

```diff
@@ -12,172 +12,309 @@
+            const auto shape = info.GetShape();
+            const auto type = info.GetElementType();
+            cv::Mat mt(std::vector<int>(shape.begin(), shape.end()), toCV(type),
+                       reinterpret_cast<void*>(result[i].GetTensorMutableData<uint8_t*>()));
+            mt.copyTo(outs[i]);
+        }
+    }
+    // One input/output overload
+    template<typename T>
+    void infer(const cv::Mat& in, cv::Mat& out) {
+        std::vector<cv::Mat> result;
+        infer<T>({in}, result);
+        GAPI_Assert(result.size() == 1u);
+        out = result.front();
+    }
+
+    void validate() {
+        GAPI_Assert(!out_gapi.empty() && !out_onnx.empty());
+        ASSERT_EQ(out_gapi.size(), out_onnx.size());
+        const auto size = out_gapi.size();
+        for (size_t i = 0; i < size; ++i) {
+            normAssert(out_onnx[i], out_gapi[i], "Test outputs");
+        }
+    }
+
+    void useModel(const std::string& model_name) {
+        model_path = findModel(model_name);
+    }
+private:
+    Ort::Env env{nullptr};
+    Ort::MemoryInfo memory_info{nullptr};
+    Ort::AllocatorWithDefaultOptions allocator;
+    Ort::SessionOptions session_options;
+    Ort::Session session{nullptr};
+
+    std::vector<int64_t> input_node_dims;
+    std::vector<std::string> in_node_names;
+    std::vector<std::string> out_node_names;
+};
+
+class ONNXClassificationTest : public ONNXtest {
+public:
+    cv::Scalar mean, std;
+    virtual void SetUp() {
+        mean = { 0.485, 0.456, 0.406 };
+        std  = { 0.229, 0.224, 0.225 };
+    }
+
+    void preprocess(const cv::Mat& src, cv::Mat& dst) {
+        const int new_h = 224;
+        const int new_w = 224;
+        cv::Mat tmp, nmat, cvt;
+        cv::resize(src, dst, cv::Size(new_w, new_h));
+        dst.convertTo(cvt, CV_32F, 1.f / 255);
+        nmat = cvt - mean;
+        tmp = nmat / std;
+        toCHW(tmp, dst);
+        dst = dst.reshape(1, {1, 3, new_h, new_w});
+    }
+};
+
+class ONNXGRayScaleTest : public ONNXtest {
+public:
+    void preprocess(const cv::Mat& src, cv::Mat& dst) {
+        const int new_h = 64;
+        const int new_w = 64;
+        cv::Mat csc, rsc, cvt;
+        cv::cvtColor(src, csc, cv::COLOR_BGR2GRAY);
+        cv::resize(csc, rsc, cv::Size(new_w, new_h));
+        rsc.convertTo(cvt, CV_32F);
+        toCHW(cvt, dst);
+        dst = dst.reshape(1, {1, 1, new_h, new_w});
+    }
+};
+} // anonymous namespace
+
+TEST_F(ONNXClassificationTest, Infer)
+{
+    useModel("classification/squeezenet/model/squeezenet1.0-9");
+    // ONNX_API code
+    cv::Mat processed_mat;
+    preprocess(in_mat1, processed_mat);
+    infer<float>(processed_mat, out_onnx.front());
     // G_API code
     G_API_NET(SqueezNet, <cv::GMat(cv::GMat)>, "squeeznet");
     cv::GMat in;
     cv::GMat out = cv::gapi::infer<SqueezNet>(in);
     cv::GComputation comp(cv::GIn(in), cv::GOut(out));
-    // NOTE: We have to normalize U8 tensor
-    // so cfgMeanStd() is here
```

Comment:
Isn't this comment still actual? Preserve, m b ?

Reply:
Fixed. But ...

Reply:
Nothing